packages/aws-cdk/lib/cdk-toolkit.ts

```diff
@@ -480,10 +561,40 @@ export class CdkToolkit {
     return assembly.stackById(stacks.firstStack.id);
   }
 
-  private assembly(): Promise<CloudAssembly> {
-    return this.props.cloudExecutable.synthesize();
+  private assembly(cacheCloudAssembly?: boolean): Promise<CloudAssembly> {
+    return this.props.cloudExecutable.synthesize(cacheCloudAssembly);
+  }
+
+  private patternsArrayForWatch(patterns: string | string[] | undefined, rootDir: string, returnRootDirIfEmpty: boolean): string[] {
+    const patternsArray: string[] = patterns !== undefined
+      ? (Array.isArray(patterns) ? patterns : [patterns])
+      : [];
+    return patternsArray.length > 0
+      // we need to prefix each pattern from the file with the root directory, as they are supposed to be relative to it
+      ? patternsArray.map(pattern => `${rootDir}/${pattern}`)
```

Comment:
Seems weird that chokidar doesn't work relative to cwd... Do we really need absolute paths here?

Reply:
`chokidar` actually _can_ work with relative directories, however I changed them to be absolute, to implement the assumption from the RFC update (https://github.com/aws/aws-cdk-rfcs/pull/383):

> The "watch" process needs to know which files and directories to observe for changes, and which ones to ignore. You can customize these using the `"include"` and `"exclude"` sub-keys of the new `"watch"` top-level key. **Values are glob patterns that are matched _relative to the location of the `cdk.json`_ directory**.

Reply:
Can't we pass the directory of CDK.out as cwd for chokidar?

Reply:
Of course we can, but why? 🙂

We explicitly said in the RFC: " Values are glob patterns that are matched relative to the location of the `cdk.json` directory.". Why do you want to change to `cdk.out` now?

Reply:
Sorry, I meant `cdk.json`.

IIUC we just need to tell `chokidar` to use the directory of `cdk.json` as `cwd`. Then all paths are calculated as relative to this directory and there is no need to prepend `${rootDir}/` to them.

Am I missing something?

Reply:
Let me test this, and I'll report back 🙂.

Reply:
Yep, it does work. Good call, this does simplify things. I'l make the change.